### PR TITLE
Replace terrafile

### PR DESCRIPTION
**Pull request by neillturner**
## Context
Replace terrafile as it's not maintained anymore. Using simple git instead.

## Changes proposed in this pull request
- Update makefile to remove references to terrafile 
- Update makefile to add new vendor-modules target with git command to correct location for terraform code
- In global_config directory  add correct value for TERRAFORM_MODULES_TAG in config files 
- In terraform/aks/config/ directory remove terrafile files 
- In terraform/domains/environment_domains/config directory remove terrafile files 
- In terraform/domains/infrastructure/config directory remove terrafile files 
- Update .gitignore  remove bin/terrafile
-  Update  .github/workflows/lint.yml to use new vendor-modules target

## Guidance to review
i tested for each environment and to see terraform correctly get the module code 
make test terraform-plan 
make development terraform-plan
make production terraform-plan
make review terraform-plan PR_NUMBER=1234
make domains-infra-plan
make test domains-plan 
make production domains-plan
make beta domains-plan
## Checklist

- [x] I have performed a self-review of my code, including formatting and typos
- [x ] I have [cleaned the commit history](https://www.annashipman.co.uk/jfdi/good-pull-requests.html)
- [x] I have added the `Devops` label
- [x ] I have attached the pull request to the trello cardme is correct

### Link to Trello card
[Replace terrafile](https://trello.com/c/5jCq2a3p/2158-replace-terrafile)

### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Review by saliceti — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -159,9 +158,13 @@ deploy-domain-arm-resources: domains domains-composed-variables arm-deployment #
 
 validate-domain-arm-resources: set-what-if domains domains-composed-variables arm-deployment ## Validate what Azure resources will be deployed. Usage: make validate-domain-arm-resources
 
-domains-infra-init: bin/terrafile domains-composed-variables set-azure-account
-	./bin/terrafile -p terraform/domains/infrastructure/vendor/modules -f terraform/domains/infrastructure/config/zones_Terrafile
+.PHONY: vendor-domain-infra-modules
+vendor-domain-infra-modules:
+	rm -rf terraform/domains/infrastructure/environment_domains/vendor/modules/domains
+	TERRAFORM_MODULES_TAG=stable
+	git clone --depth=1 --single-branch --branch ${TERRAFORM_MODULES_TAG} https://github.com/DFE-Digital/terraform-modules.git terraform/domains/infrastructure/vendor/modules/domains
```
Comment (saliceti) [inline Makefile:165]: Can we write it `git -c advice.detachedHead=false clone...` to suppress the warning message

File: Makefile [hunk 2]
```diff
@@ -173,9 +176,12 @@ domains-infra-plan: domains domains-infra-init ## Terraform plan for DNS infrast
 domains-infra-apply: domains domains-infra-init ## Terraform apply for DNS infrastructure (zone and front door). Usage: make domains-infra-apply
 	terraform -chdir=terraform/domains/infrastructure apply -var-file config/zones.tfvars.json ${AUTO_APPROVE}
 
-domains-init: bin/terrafile domains-composed-variables set-azure-account
-	./bin/terrafile -p terraform/domains/environment_domains/vendor/modules -f terraform/domains/environment_domains/config/${ENVIRONMENT}_Terrafile
+.PHONY: vendor-domain-modules
+vendor-domain-modules:
+	rm -rf terraform/domains/environment_domains/vendor/modules/domains
+	git clone --depth=1 --single-branch --branch ${TERRAFORM_MODULES_TAG} https://github.com/DFE-Digital/terraform-modules.git terraform/domains/environment_domains/vendor/modules/domains
```
Comment (saliceti) [inline Makefile:182]: Can we write it `git -c advice.detachedHead=false clone...` to suppress the warning message

File: Makefile [hunk 3]
```diff
@@ -91,19 +90,19 @@ composed-variables:
 	$(eval STORAGE_ACCOUNT_NAME=${AZURE_RESOURCE_PREFIX}${SERVICE_SHORT}${CONFIG_SHORT}tfsa)
 	$(eval LOG_ANALYTICS_WORKSPACE_NAME=${AZURE_RESOURCE_PREFIX}-${SERVICE_SHORT}-${CONFIG_SHORT}-log)
 
-bin/terrafile: ## Install terrafile to manage terraform modules
-	curl -sL https://github.com/coretech/terrafile/releases/download/v${TERRAFILE_VERSION}/terrafile_${TERRAFILE_VERSION}_$$(uname)_x86_64.tar.gz \
-		| tar xz -C ./bin terrafile
+.PHONY: vendor-modules
+vendor-modules:
+	rm -rf terraform/aks/vendor/modules/aks
+	git clone --depth=1 --single-branch --branch ${TERRAFORM_MODULES_TAG} https://github.com/DFE-Digital/terraform-modules.git terraform/aks/vendor/modules/aks
```
[the remark this answers is not in the document]
Comment (neillturner) [reply Makefile]: fixed

File: Makefile (hunk 1 above)
Comment (neillturner) [reply Makefile]: fixed

File: Makefile (hunk 2 above)
Comment (neillturner) [reply Makefile]: fixed

**Review by saliceti — APPROVED** [review]

**Pull request merged**